### PR TITLE
Use stable version of classification-bundle

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is for next major.

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Psalm failing

**Comment by jordisala1991** [comment]
> Psalm failing

Look at this commit: https://github.com/sonata-project/SonataMediaBundle/pull/2256/commits/2b8e7fedfd0efaadbc72fcff6b02200f0675f1cd

I don't think this should be needed, but for some reason it is not working without it. It detects scalar instead of string. I can't use assert because phpstan complains.

**Comment by VincentLanglet** [comment]
I'd like to investigate a little more before adding some psalm-suppress.

The code was ok on the previous PR, we should look which dependency is the reason of this failure

**Comment by VincentLanglet** [comment]
Ok, I find the issue: it's https://github.com/psalm/psalm-plugin-symfony/issues/233

**Review by core23 — DISMISSED** [review]

**Review by VincentLanglet — CHANGES_REQUESTED** [review]
Can you try again without the psalm-suppress ?

The 3.1.1 version of the psalm plugin is released.

**Comment by VincentLanglet** [comment]
> Can you try again without the psalm-suppress ?
> 
> The 3.1.1 version of the psalm plugin is released.

I tried locally and I think https://github.com/psalm/psalm-plugin-symfony/pull/235 we'll be needed

File: src/Admin/BaseMediaAdmin.php [hunk 1]
```diff
@@ -83,23 +83,28 @@ protected function configurePersistentParameters(): array
...
         if (\is_array($filter) && \array_key_exists('context', $filter)) {
             $context = $filter['context']['value'];
         } else {
-            $context = $this->getRequest()->query->get('context', $this->pool->getDefaultContext());
+            $context = $request->query->get('context', $this->pool->getDefaultContext());
         }
 
+        /**
+         * @psalm-suppress PossiblyNullArgument
+         */
```
Comment (VincentLanglet) [inline src/Admin/BaseMediaAdmin.php]:
```suggestion
```

File: src/Admin/BaseMediaAdmin.php [hunk 2]
```diff
@@ -122,42 +127,57 @@ protected function configurePersistentParameters(): array
...
+        $context = $request->query->get('context');
 
-                if (null !== $categoryContext && $categoryContext->getId() === $context) {
-                    $object->setCategory($category);
-                }
-            }
+        $object->setProviderName($providerName);
+        /**
+         * @psalm-suppress InvalidArgument
+         */
```
Comment (VincentLanglet) [inline src/Admin/BaseMediaAdmin.php]:
```suggestion
```

File: src/Admin/GalleryAdmin.php [hunk 3]
```diff
@@ -65,9 +65,14 @@ protected function configurePersistentParameters(): array
...
     {
-        if ($this->hasRequest()) {
-            $object->setContext($this->getRequest()->query->get('context'));
+        if (!$this->hasRequest()) {
+            return;
         }
+
+        /**
+         * @psalm-suppress InvalidArgument
+         */
```
Comment (VincentLanglet) [inline src/Admin/GalleryAdmin.php]:
```suggestion
```

File: src/Controller/GalleryAdminController.php [hunk 4]
```diff
@@ -39,6 +39,9 @@ public function listAction(Request $request): Response
 
         $listMode = $request->query->get('_list_mode');
         if (null !== $listMode) {
+            /**
+             * @psalm-suppress InvalidArgument
+             */
```
Comment (VincentLanglet) [inline src/Controller/GalleryAdminController.php]:
```suggestion
```

File: src/Controller/MediaAdminController.php [hunk 5]
```diff
@@ -43,6 +43,9 @@ public function createAction(Request $request): Response
             $pool = $this->container->get('sonata.media.pool');
             \assert($pool instanceof Pool);
 
+            /**
+             * @psalm-suppress InvalidArgument
+             */
```
Comment (VincentLanglet) [inline src/Controller/MediaAdminController.php]:
```suggestion
```

File: src/Controller/MediaAdminController.php [hunk 6]
```diff
@@ -65,6 +68,9 @@ public function listAction(Request $request): Response
             return $preResponse;
         }
 
+        /**
+         * @psalm-suppress InvalidArgument
+         */
```
Comment (VincentLanglet) [inline src/Controller/MediaAdminController.php]:
```suggestion
```

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Comment by VincentLanglet** [comment]
Is it ready for a 4.0.0 release ? @jordisala1991 

Edit: maybe https://github.com/sonata-project/SonataMediaBundle/issues/2232 need to be solved ?

**Comment by jordisala1991** [comment]
> Is it ready for a 4.0.0 release ? @jordisala1991 
> 
> 
> 
> Edit: maybe https://github.com/sonata-project/SonataMediaBundle/issues/2232 need to be solved ?

It would be nice to solve that one indeed. Besides from that, we are ready.

**Comment by VincentLanglet** [comment]
Did you try the 4.x branch ? @jordisala1991
If it works for you I think we should release the 4.0.0 version even if there is https://github.com/sonata-project/SonataMediaBundle/issues/2232

**Pull request merged**